packages/mdc-list/README.md

```diff
@@ -200,30 +239,76 @@ Key | Action
 `ArrowRight` | When the list is in a horizontal orientation (default), it will cause the next list item to receive focus.
 `Home` | Will cause the first list item in the list to receive focus.
 `End` | Will cause the last list item in the list to receive focus.
+`Space` | Will cause the currently focused list item to become selected/deselected if `singleSelection=true`.
+`Enter` | Will cause the currently focused list item to become selected/deselected if `singleSelection=true`.
 
 ## Usage within Web Frameworks
 
 If you are using a JavaScript framework, such as React or Angular, you can create a List for your framework. Depending on your needs, you can use the _Simple Approach: Wrapping MDC Web Vanilla Components_, or the _Advanced Approach: Using Foundations and Adapters_. Please follow the instructions [here](../../docs/integrating-into-frameworks.md).
 
+### Considerations for Advanced Approach
+
+The `MDCListFoundation` expects the HTML to be setup a certain way before being used. This setup is a part of the `layout()` and `singleSelection()` functions within the `index.js`.
+
+#### Setup in `layout()`
+
+The default component requires that every list item receives a `tabindex` value so that it can receive focus
+(`li` elements cannot receive focus at all without a `tabindex` value). Any element not already containing a
+`tabindex` attribute will receive `tabindex=-1`. The first list item should have `tabindex="0"` so that the
+user can find the first element using the `tab` key, but subsequent `tab` keys strokes will cause focus to
+skip over the entire list. If the list items contain sub-elements that are focusable (`button` or `a` elements), 
+these should also receive `tabIndex="-1"`.
+
+```html
+<ul id="my-list" class="mdc-list" aria-orientation="vertical">
+  <li class="mdc-list-item" tabindex="0">Single-line item<button tabindex="-1"></button></li>
+  <li class="mdc-list-item" tabindex="-1">Single-line item</li>
+  <li class="mdc-list-item" tabindex="-1">Single-line item</li>
+</ul>
+```
+
+#### Setup in `singleSelection()`
+
+When implementing a component that will use the single selection variant, the HTML should be modified to include
+the `aria-selected` attribute, the `mdc-list-item--selected` class should be added, and the `tabindex` of the selected
+element should be `0`. The first list item should have the `tabindex` updated to `-1`. The foundation method 
+`setSelectedIndex()` should be called with the initially selected element immediately after the foundation is 
+instantiated.
+
+```html
+<ul id="my-list" class="mdc-list" aria-orientation="vertical">
+  <li class="mdc-list-item" tabindex="-1">Single-line item<button tabindex="-1"></button></li>
+  <li class="mdc-list-item mdc-list-item--selected" aria-selected="true" tabindex="0">Single-line item</li>
+  <li class="mdc-list-item" tabindex="-1">Single-line item</li>
+</ul>
+```
```

Comment:
I think adding the JS here showing to call `setSelectedIndex` will complete the documentation....if I'm reading that correctly.

Reply:
The documentation shows the JS in above examples. This example is discussing how the html should be rendered when creating your own component.